### ISSUE TITLE
App crashes if "REST API Upload" is switched on, but "Nightscout URL" is invalid.

### ISSUE THREAD

**Issue by pazaan** [pazaan/640gAndroidUploader#37]
http://{YOUR-API-SERVER} will cause the app to crash with this:

http://{YOUR-API-SERVER}/api/devicestatus
    at java.net.URI.create(URI.java:733) ~[na:0.0]
    at org.apache.http.client.methods.HttpPost.<init>(HttpPost.java:84) ~[na:0.0]
    at info.nightscout.android.upload.UploadHelper.postDeviceStatus(UploadHelper.java:694) ~[na:0.0]
    at info.nightscout.android.upload.UploadHelper.doRESTUploadTo(UploadHelper.java:481) ~[na:0.0]
    at info.nightscout.android.upload.UploadHelper.doRESTUpload(UploadHelper.java:417) ~[na:0.0]
    at info.nightscout.android.upload.UploadHelper.doInBackground(UploadHelper.java:364) ~[na:0.0]
    at info.nightscout.android.upload.UploadHelper.doInBackground(UploadHelper.java:65) ~[na:0.0]
    at android.os.AsyncTask$2.call(AsyncTask.java:295) ~[na:0.0]
    at java.util.concurrent.FutureTask.run(FutureTask.java:237) ~[na:0.0]
    at android.os.AsyncTask$SerialExecutor$1.run(AsyncTask.java:234) ~[na:0.0]
    at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1113) ~[na:0.0]
    at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:588) ~[na:0.0]
    at java.lang.Thread.run(Thread.java:818) ~[na:0.0]

**Comment by jgwashburn** [comment]
I took a look at this to see if it is still occurring in the latest code version. (BTW it turns out that relevant code section has been rewritten since the bug was filed , to the extent that the stack trace above is no longer applicable).

 I tried to reproduce the error by setting an invalid Nightscout URL on the phone, and running the app with the debugger. What I find is that in that case the code catches the resulting exception, at 
https://github.com/pazaan/640gAndroidUploader/blob/63e1bdb8cfda52d172a6dcca9254a4fb6ad8a204/app/src/main/java/info/nightscout/android/upload/nightscout/NightscoutUploadIntentService.java#L239

and hence no crash is seen.

Does anyone have an objection to closing this issue as fixed?

Thanks
